src/seedu/addressbook/data/person/Name.java

```diff
@@ -37,10 +38,15 @@ public static boolean isValidName(String test) {
     }
 
     /**
-     * Retrieves a listing of every word in the name, in order.
+     * Retrieves a listing of every word in the name, in order and in lowercase.
      */
-    public List<String> getWordsInName() {
-        return Arrays.asList(fullName.split("\\s+"));
+    public List<String> getWordsInNameInLowerCase() {
+        String[] wordsInName = fullName.split("\\s+");
+        List<String> lowerCaseWordsInName = new LinkedList<String>();
```

Comment:
Interesting! But why do we need to use a LinkedList here, shouldn't a List suffice? Especially since we (probably) aren't dealing with too many entries in the address book.

Reply:
I implemented it this way because List is a interface and LinkedList is a class inplementing this interface. This was the best way i could think of, is there a better way?

Reply:
The existing Arrays.asList is one way to do this. You could also convert the name to lowercase first, then let Arrays.asList handle the now lowercase'd name.